src/main.rs

```diff
@@ -734,15 +762,40 @@ async fn process_signed_delegate_action(
     if USE_PAY_WITH_FT.clone() {
         let non_delegate_actions = signed_delegate_action.delegate_action.get_actions();
         let treasury_payments: Vec<Action> = non_delegate_actions
-            .into_iter()
-            .filter(|x| matches!(
-                x,
-                Action::FunctionCall(FunctionCallAction { args, .. }
-                ) if String::from_utf8_lossy(args).contains(&BURN_ADDRESS.to_string()))
-            )
+            .iter()
+            .filter_map(|action| {
+                if let Action::FunctionCall(FunctionCallAction { args, .. }) = action {
+                    debug!("args: {:?}", args);
+
+                    // convert to ascii lowercase
+                    let args_ascii = args.to_ascii_lowercase();
+                    debug!("args_ascii: {:?}", args_ascii);
+
+                    // Convert to UTF-8 string
+                    let args_str = String::from_utf8_lossy(&args_ascii);
+                    debug!("args_str: {:?}", args_str);
+
+                    // Parse to JSON (assuming args are serialized as JSON)
+                    let args_json: Value =
+                        serde_json::from_str(&args_str).expect("Failed to parse JSON");
+                    debug!("args_json: {:?}", args_json);
+
+                    // get the receiver_id from the json without the escape chars
+                    let receiver_id = args_json["receiver_id"].as_str().unwrap_or_default();
+                    debug!("receiver_id: {receiver_id}");
+                    debug!("BURN_ADDRESS.to_string(): {:?}", BURN_ADDRESS.to_string());
+
+                    // Check if receiver_id in args contain BURN_ADDRESS
+                    if receiver_id == BURN_ADDRESS.to_string() {
+                        debug!("SignedDelegateAction contains the BURN_ADDRESS MATCH");
+                        return Some(action.clone());
+                    }
+                }
+                None
+            })
```

Comment:
This looks pretty funky way of deserializing the args. What's the expected type here and why not directly deserialize into that type over doing this manual way of checking for receiver id? Also, what's the `to_ascii_lowercase` for? Seems weird to have it since the action should not have any sort of uppercasing when it comes to account ids, since they're constrained to lowercase by default anyways.

Reply:
This is deserializing the `args` from json. I convert to ascii lowercase to make any comparisons case insensitive. The raw payload of args is base64 encoded within the json. The deserialization of args is necessary bc the args comes in a a vec<u8> (initial deserialization of json payload already happened).

Reply:
> I convert to ascii lowercase to make any comparisons case insensitive.

Then shouldn't be necessary then since account/contract addresses are always lowercase no matter what. Passing them as uppercase will fail on contract runtime. If anything, we should actually fail early here in the relayer and alert that the receiver id is in uppercase and not in the correct format. You can just do a `AccountId::parse` on it to validate this.

So this could then be reduced to something like this utilizing the `serde_json::from_slice` since there's no point converting it to a string:
```rs
let args: Value = serde_json::from_slice(args).unwrap_or_else(|err| {
    Value::Null
});
let receiver_id = args_json["receiver_id"].as_str().unwrap_or_default();
if receiver_id == BURN_ADDRESS.to_string() {
    return Some(action);
}
```